### DISCUSSION TITLE
Make keybinding for vi command mode

### DISCUSSION THREAD

**Discussion by rosshadden** [nushell/nushell#4252] [Q&A] [answered]
Is there a way to specify a mode for a keybinding? For example I want to map some keys in the vim command mode that wouldn't make sense to map in insert mode (which is where the mappings go by default).

Something like this, to get the idea across:

```config
# newline
- key:
    Char: o
  mode: vi-command
  binding:
    NewLine:
```

Unfortunately `NewLine` does not take a direction argument so I don't think I could make the `o` and `O` mappings I was hoping to. I use them a lot in zsh's vim mode to easily create multiline commands while in the shell.

**Comment by fdncred** [comment, marked as answer]
i don't think you can assign keybindings to a mode in nushell currently but we're working on keybindings in our latest engine, called engine-q. there's already and emacs mode and a vi mode. In nushell, all you have is the `config.toml` and the setting that says `edit_mode = "emacs" # or "vi"`. I think this is a good add for engine-q though. Woud you mind creating an issue over at https://github.com/nushell/engine-q that describes the functionality you're looking for?

**Reply by rosshadden** [reply]
Sure thing, I'll do that.

Update: Created https://github.com/nushell/engine-q/issues/727
